File: Bindings~/java/src/main/java/com/alttester/Commands/UnityCommand/AltGetAllLoadedScenes.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.UnityCommand;

import com.alttester.AltMessage;
import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltGetAllLoadedScenes extends AltBaseCommand {
  public AltGetAllLoadedScenes(IMessageHandler messageHandler) {
    super(messageHandler);
  }

  public String[] Execute() {
    AltMessage altMessage = new AltMessage();
    altMessage.setCommandName("getAllLoadedScenes");
    SendCommand(altMessage);
    return recvall(altMessage, String[].class);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/UnityCommands/AltGetAllLoadedScenes.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

using System.Collections.Generic;

namespace AltTester.AltTesterSDK.Driver.Commands
{
    internal class AltGetAllLoadedScenes : AltBaseCommand
    {
        private readonly AltGetAllLoadedScenesParams cmdParams;
        public AltGetAllLoadedScenes(IDriverCommunication commHandler) : base(commHandler)
        {
            cmdParams = new AltGetAllLoadedScenesParams();
        }
        public List<string> Execute()
        {
            CommHandler.Send(cmdParams);
            return CommHandler.Recvall<List<string>>(cmdParams);
        }
    }
}
